lib/fluent/plugin/elasticsearch_error_handler.rb

```diff
@@ -4,20 +4,22 @@ class Fluent::ElasticsearchErrorHandler
   include Fluent::ElasticsearchConstants
 
   attr_accessor :records, :bulk_message_count
-  class BulkIndexQueueFull < StandardError; end
-  class ElasticsearchOutOfMemory < StandardError; end
   class ElasticsearchVersionMismatch < StandardError; end
-  class UnrecognizedElasticsearchError < StandardError; end
   class ElasticsearchError < StandardError; end
+
   def initialize(plugin, records = 0, bulk_message_count = 0)
     @plugin = plugin
     @records = records
     @bulk_message_count = bulk_message_count
   end
 
-  def handle_error(response)
+  def handle_error(response, tag, records)
```

Comment:
pass `chunk` instead of `records`

Reply:
We can not pass chunk instead of records.  This code is executed after we have deserialized the individual records in the chunk to confirm they dont throw exceptions like 'utf-8' problems.  The chunk records are not representative of those we submitted to Elasticsearch